wgpu-rs/src/lib.rs

```diff
@@ -233,6 +233,29 @@ impl<'a> TextureCopyView<'a> {
     }
 }
 
+pub struct CreateBufferMapped<'a, T> {
+    id: wgn::BufferId,
+    data: &'a mut [T],
+}
+
+impl<'a, T> CreateBufferMapped<'a, T>
+        where T: Copy {
+    pub fn from_slice(self, slice: &[T]) -> Buffer {
+        self.data.copy_from_slice(slice);
+        self.finish()
+    }
+
+    pub fn as_mut_slice<'b>(&'b mut self) -> &'b mut [T] {
```

Comment:
Rust conventions say we should rename this to `as_slice_mut`. Also, let's omit the lifetimes - they are derived automatically here.

Reply:
also... why do we need a method here at all if we can just map `data` public? :)